biogrid/libs/biogrid-simulator/src/lib/biobattery/biobattery.spec.ts

```diff
@@ -1,41 +1,82 @@
-import { BioBattery } from "./";
-import { RESISTANCE } from "../config";
+import { BioBattery } from './';
+import { RESISTANCE } from '../config';
 
 describe('test for BioBattery class', () => {
   test('battery object throws an error if passed negative Energy', () => {
     const currentpower = -1;
     const maxCapacity = 5;
-    expect(() => new BioBattery(10, 5,  'small_battery', RESISTANCE.SMALL_BATTERY, currentpower, maxCapacity)).toThrow(
+    expect(
+      () =>
+        new BioBattery({
+          xPos: 10,
+          yPos: 5,
+          id: 'small_battery',
+          resistance: RESISTANCE.SMALL_BATTERY,
+          initialEnergyInJoules: currentpower,
+          maxCapacityInJoules: maxCapacity,
+        })
+    ).toThrow(
       `Cannot create a battery with values: (${currentpower}, ${maxCapacity})`
     );
   });
 
   test('battery object throws an error if passed energy more than its capacity', () => {
     const currentpower = 10;
     const maxCapacity = 1;
-    expect(() => new BioBattery(10, 5,  'small_battery', RESISTANCE.SMALL_BATTERY, currentpower, maxCapacity)).toThrow(
+    expect(
+      () =>
+        new BioBattery({
+          xPos: 10,
+          yPos: 5,
+          id: 'small_battery',
+          resistance: RESISTANCE.SMALL_BATTERY,
+          initialEnergyInJoules: currentpower,
+          maxCapacityInJoules: maxCapacity,
+        })
+    ).toThrow(
       `Cannot create a battery with values: (${currentpower}, ${maxCapacity})`
     );
   });
 
   test('battery returns less or equal to power requested', () => {
     const currentpower = 20;
     const maxCapacity = 30;
-    const battery = new BioBattery(10, 5,  'small_battery', RESISTANCE.SMALL_BATTERY, currentpower, maxCapacity);
+    const battery = new BioBattery({
+      xPos: 10,
+      yPos: 5,
+      id: 'small_battery',
+      resistance: RESISTANCE.SMALL_BATTERY,
+      initialEnergyInJoules: currentpower,
+      maxCapacityInJoules: maxCapacity,
+    });
     const outputenergy = 25;
     const expected = 25;
     expect(battery.supplyPower(outputenergy)).toBeLessThanOrEqual(expected);
```

Comment:
I know this isn't part of your PR but why is it lessthanorequal rather than exactly the expected value?

Reply:
@RolandNaijuka, correct me if I am wrong, I believe it has to do with efficiency and losses from delivering the energy. So, not all outputed energy is delivered

Reply:
That is right